packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -150,26 +150,25 @@ export class RelationshipsController extends TransportController {
         return await CoreCrypto.verify(content, signature, relationship.peer.publicKey);
     }
 
-    public async sendRelationship(parameters: ISendRelationshipParameters): Promise<Relationship> {
-        parameters = SendRelationshipParameters.from(parameters);
-        const template = (parameters as SendRelationshipParameters).template;
-        if (!template.cache) {
-            throw this.newCacheEmptyError(RelationshipTemplate, template.id.toString());
-        }
+    public async sendRelationship(parameters: ISendRelationshipParameters): Promise<Result<Relationship>> {
+        const canSendRelationship = await this.canSendRelationship(parameters);
 
-        const queryForExistingRelationships = {
-            "peer.address": template.cache.createdBy.toString(),
-            status: { $in: [RelationshipStatus.Pending, RelationshipStatus.Active, RelationshipStatus.Terminated, RelationshipStatus.DeletionProposed] }
-        };
+        if (!canSendRelationship.isSuccess) {
```

Comment:
Only a question: Wouldn't it be easier to use 
`if (!canSendRelationship.isSuccess && canSendRelationship.error.code !== "error.transport.relationships.relationshipNotYetDecomposedByPeerOrPeerIsToBeDeleted")
{throw canSendRelationship.error}`
and after that handle the other case ?

Reply:
I'd go even further that this returning a result in one case and throwing in the other is messy and should be unified

Reply:
No I meant what finally changed, so I resolve this conversation.